### ISSUE TITLE
Missing denotation for case class unapply

### ISSUE THREAD

**Issue by olafurpg** [scalameta/scalameta#830]
This code here

```scala
object A {
  def unapply(e: Int): Option[String] = Some(e.toString)
}
case class Foo(a: Int)
object ya {
  A.unapply(2)
  Foo.unapply(Foo(2))
}
```

has database

```

Names:
[7..8): A => _empty_.A.
[17..24): unapply => _empty_.A.unapply(I)Lscala/Option;.
[25..26): e => _empty_.A.unapply(I)Lscala/Option;.(e)
[28..31): Int => _root_.scala.Int#
[34..40): Option => _root_.scala.Option#
[41..47): String => _root_.scala.Predef.String#
[51..55): Some => _root_.scala.Some.apply(Ljava/lang/Object;)Lscala/Some;.
[56..57): e => _empty_.A.unapply(I)Lscala/Option;.(e)
[58..66): toString => _root_.scala.Any#toString()Ljava/lang/String;.
[81..84): Foo => _empty_.Foo#
[85..86): a => _empty_.Foo#(a)
[88..91): Int => _root_.scala.Int#
[100..102): ya => _empty_.ya.
[107..108): A => _empty_.A.
[109..116): unapply => _empty_.A.unapply(I)Lscala/Option;.
[122..125): Foo => _empty_.Foo.
[126..133): unapply => _empty_.Foo.unapply(LFoo;)Lscala/Option;.
[134..137): Foo => _empty_.Foo.apply(I)LFoo;.

Denotations:
_empty_.A. => final object A
_empty_.A.unapply(I)Lscala/Option;. => def unapply: (e: Int)Option[String]
_empty_.A.unapply(I)Lscala/Option;.(e) => param e: Int
_empty_.Foo# => case class Foo
_empty_.Foo#(a) => val param a: Int
_empty_.Foo#`<init>`(I)V. => primaryctor <init>: (a: Int)Foo
_empty_.ya. => final object ya

Sugars:
[55..55) [String]
```

Observe that `A.unapply` has a denotation but `Foo.unapply` does not. It would be nice if we could get the denotation for the compiler generated `Foo.unapply`

**Comment by olafurpg** [comment]
This opens up a question whether we should persist denotations of synthetic definitions. The members of the companion module are quite many
```scala
gsym.companionModule.toType.members]:
Scope{
  private def readResolve(): Object;
  case def unapply(x$0: Foo): Option[Int];
  case def apply(a: Int): Foo;
  final override def toString(): String;
  def <init>(): Foo.type;
  def andThen: <?>;
  def compose: <?>;
  def $init$: <?>;
  final def $asInstanceOf[T0](): T0;
  final def $isInstanceOf[T0](): Boolean;
  final def synchronized[T0](x$1: T0): T0;
  final def ##(): Int;
  final def !=(x$1: Any): Boolean;
  final def ==(x$1: Any): Boolean;
  final def ne(x$1: AnyRef): Boolean;
  final def eq(x$1: AnyRef): Boolean;
  protected[package lang] def finalize(): Unit;
  final def wait(): Unit;
  final def wait(x$1: Long,x$2: Int): Unit;
  final def wait(x$1: Long): Unit;
  final def notifyAll(): Unit;
  final def notify(): Unit;
  protected[package lang] def clone(): Object;
  def equals(x$1: Any): Boolean;
  def hashCode(): Int;
  final def getClass(): Class[_];
  final def asInstanceOf[T0]: T0;
  final def isInstanceOf[T0]: Boolean
}
```
I believe the ones with the `case` flag are the most interesting ones and should be included in the semantic DB. WDYT @xeno-by ?

**Comment by olafurpg** [comment]
This got fixed in #939. I added a test to double-check it's fixed https://github.com/scalameta/scalameta/pull/983

**Comment by olafurpg** [comment]
Fixed in #988
